Server routes order

Hi,
Thanks for the great provider, I've been dreaming to have it for such a long time.
After importing existing server as `pritunl_server` resource, `terraform_plan` suggests multiple changes to the routes which looks like something related to routes ordering:

```
resource "pritunl_server" "my-vpn" {
    name    = "my-vpn"
    port    = 1196
    network = "10.8.35.0/25"
  
    organization_ids = [
      pritunl_organization.this_orgs["_tools_"].id
    ]

    route {
      network = "10.1.0.0/16"
      comment = "route-1"
      nat     = true
    }
    route {
      network = "10.2.4.234/32"
      comment = "route-2"
      nat     = true
    }  
    route {
      network = "10.4.0.0/16"
      comment = "route-3"
      nat     = true
    }
  }

$terraform plan --target pritunl_server.my-vpn
pritunl_server.my-vpn: Refreshing state... [id=5e45018371f6b2996983beac]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # pritunl_server.my-vpn will be updated in-place
  ~ resource "pritunl_server" "my-vpn" {
        id                 = "5e45018371f6b2996983beac"
        name               = "my-vpn"
        # (31 unchanged attributes hidden)

      ~ route {
          ~ comment = "route-2" -> "route-1"
          ~ network = "10.2.4.234/32" -> "10.1.0.0/16"
            # (1 unchanged attribute hidden)
        }
      ~ route {
          ~ comment = "route-3" -> "route-2"
          ~ network = "10.4.0.0/16" -> "10.2.4.234/32"
            # (1 unchanged attribute hidden)
        }
      ~ route {
          ~ comment = "route-1" -> "route-3"
          ~ network = "10.1.0.0/16" -> "10.4.0.0/16"
            # (1 unchanged attribute hidden)
        }
    }
```

Pritunl API returns data in the same order as stated in `pritunl_server` resource :
'GET',  '/server/5e45018371f6b2996983beac/route'

```
[
   {
      "comment":"None",
      "nat_netmap":"None",
      "network":"10.8.35.0/25",
      "vpc_region":"None",
      "net_gateway":false,
      "network_link":false,
      "metric":"None",
      "server":"5e45018371f6b2996983beac",
      "nat":false,
      "virtual_network":true,
      "vpc_id":"None",
      "advertise":"None",
      "link_virtual_network":false,
      "nat_interface":"None",
      "id":"31302e382e33352e302f3235",
      "server_link":false
   },
   {
      "comment":"route-1",
      "nat_netmap":"None",
      "network":"10.1.0.0/16",
      "vpc_region":"None",
      "net_gateway":false,
      "network_link":false,
      "metric":"None",
      "server":"5e45018371f6b2996983beac",
      "nat":true,
      "virtual_network":false,
      "vpc_id":"None",
      "advertise":false,
      "link_virtual_network":false,
      "nat_interface":"None",
      "id":"31302e312e302e302f3136",
      "server_link":false
   },
   {
      "comment":"route-2",
      "nat_netmap":"None",
      "network":"10.2.4.234/32",
      "vpc_region":"None",
      "net_gateway":false,
      "network_link":false,
      "metric":"None",
      "server":"5e45018371f6b2996983beac",
      "nat":true,
      "virtual_network":false,
      "vpc_id":"None",
      "advertise":false,
      "link_virtual_network":false,
      "nat_interface":"None",
      "id":"31302e322e342e3233342f3332",
      "server_link":false
   },
   {
      "comment":"route-3",
      "nat_netmap":"None",
      "network":"10.4.0.0/16",
      "vpc_region":"None",
      "net_gateway":false,
      "network_link":false,
      "metric":"None",
      "server":"5e45018371f6b2996983beac",
      "nat":true,
      "virtual_network":false,
      "vpc_id":"None",
      "advertise":false,
      "link_virtual_network":false,
      "nat_interface":"None",
      "id":"31302e342e302e302f3136",
      "server_link":false
   }
]
```
Is there anything to do with this? Thanks!

FWIW, I run into the ordering of routes intermittently running the test make target:

```
❯ make test
Error: No such container: tf_pritunl_acc_test
[REDACTED]
sleep 10
./tools/wait-for-it.sh localhost:27017 -- echo "mongodb is up"
wait-for-it.sh: waiting 15 seconds for localhost:27017
wait-for-it.sh: localhost:27017 is available after 0 seconds
mongodb is up
# enables an api access for the pritunl user, updates an api token and secret
MongoDB shell version: 3.2.22
connecting to: test
switched to db pritunl
WriteResult({ "nMatched" : 1, "nUpserted" : 0, "nModified" : 1 })
bye
TF_ACC=1 \
	PRITUNL_URL="https://localhost/" \
	PRITUNL_INSECURE="true" \
	PRITUNL_TOKEN=tfacctest_token \
	PRITUNL_SECRET=tfacctest_secret \
	go test -v -cover -count 1 ./internal/provider
=== RUN   TestDataSourceHost
=== PAUSE TestDataSourceHost
=== RUN   TestAccOrganization_basic
=== PAUSE TestAccOrganization_basic
=== RUN   TestGetServer_basic
--- PASS: TestGetServer_basic (50.69s)
=== RUN   TestGetServer_with_attached_organization
--- PASS: TestGetServer_with_attached_organization (33.14s)
=== RUN   TestGetServer_with_a_few_attached_organizations
--- PASS: TestGetServer_with_a_few_attached_organizations (33.44s)
=== RUN   TestGetServer_with_attached_route
    resource_server_test.go:166: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.

        (map[string]string) (len=1) {
         (string) (len=6) "status": (string) (len=7) "offline"
        }


        (map[string]string) (len=1) {
         (string) (len=6) "status": (string) (len=7) "pending"
        }
--- FAIL: TestGetServer_with_attached_route (31.62s)
=== RUN   TestGetServer_with_a_few_attached_routes
    resource_server_test.go:216: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.

        (map[string]string) (len=2) {
         (string) (len=15) "route.0.network": (string) (len=11) "10.3.0.0/24",
         (string) (len=15) "route.1.network": (string) (len=11) "10.2.0.0/24"
        }


        (map[string]string) (len=2) {
         (string) (len=15) "route.0.network": (string) (len=11) "10.2.0.0/24",
         (string) (len=15) "route.1.network": (string) (len=11) "10.3.0.0/24"
        }
--- FAIL: TestGetServer_with_a_few_attached_routes (25.03s)
=== RUN   TestGetServer_with_invalid_route
--- PASS: TestGetServer_with_invalid_route (7.08s)
=== RUN   TestCreateServer_with_invalid_network
--- PASS: TestCreateServer_with_invalid_network (9.41s)
=== RUN   TestCreateServer_with_unsupported_network
--- PASS: TestCreateServer_with_unsupported_network (21.67s)
=== RUN   TestCreateServer_with_invalid_bind_address
--- PASS: TestCreateServer_with_invalid_bind_address (20.93s)
=== RUN   TestGetServer_with_default_host
--- PASS: TestGetServer_with_default_host (19.15s)
=== RUN   TestGetServer_without_hosts
--- PASS: TestGetServer_without_hosts (32.28s)
=== RUN   TestAccUser_basic
=== PAUSE TestAccUser_basic
=== CONT  TestDataSourceHost
=== CONT  TestAccUser_basic
=== CONT  TestAccOrganization_basic
--- PASS: TestDataSourceHost (24.42s)
--- PASS: TestAccOrganization_basic (54.67s)
--- PASS: TestAccUser_basic (56.05s)
FAIL
coverage: 64.3% of statements
FAIL	github.com/disc/terraform-provider-pritunl/internal/provider	340.730s
FAIL
make: *** [test] Error 1
```

--- a/internal/provider/resource_server.go
+++ b/internal/provider/resource_server.go
@@ -528,7 +528,12 @@ func resourceReadServer(ctx context.Context, d *schema.ResourceData, meta interf
 		if !ok {
 			return diag.Errorf("failed to parse organization_ids for the server: %s", server.Name)
 		}
-		d.Set("organization_ids", matchStringEntitiesWithSchema(organizationsList, declaredOrganizations))
+
+		if len(declaredOrganizations) > 0 {
+			organizationsList = matchStringEntitiesWithSchema(organizationsList, declaredOrganizations)
+		}
+
+		d.Set("organization_ids", organizationsList)
 	}
 
 	if len(server.Groups) > 0 {
@@ -542,7 +547,12 @@ func resourceReadServer(ctx context.Context, d *schema.ResourceData, meta interf
 		if !ok {
 			return diag.Errorf("failed to parse groups for the server: %s", server.Name)
 		}
-		d.Set("groups", matchStringEntitiesWithSchema(groupsList, declaredGroups))
+
+		if len(declaredGroups) > 0 {
+			groupsList = matchStringEntitiesWithSchema(groupsList, declaredGroups)
+		}
+
+		d.Set("groups", groupsList)
 	}
 
 	if len(routes) > 0 {
@@ -550,7 +560,12 @@ func resourceReadServer(ctx context.Context, d *schema.ResourceData, meta interf
 		if !ok {
 			return diag.Errorf("failed to parse routes for the server: %s", server.Name)
 		}
-		d.Set("route", flattenRoutesData(matchRoutesWithSchema(routes, declaredRoutes)))
+
+		if len(declaredRoutes) > 0 {
+			routes = matchRoutesWithSchema(routes, declaredRoutes)
+		}
+
+		d.Set("route", flattenRoutesData(routes))
 	}
 
 	if len(hosts) > 0 {

